examples/cpp/security/CLIParser.hpp

```diff
@@ -0,0 +1,292 @@
+// Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <csignal>
+#include <cstdlib>
+#include <iostream>
+#include <cmath>
+
+#include <fastdds/dds/log/Log.hpp>
+
+#ifndef FASTDDS_EXAMPLES_CPP_SECURITY__CLI_PARSER_HPP
+#define FASTDDS_EXAMPLES_CPP_SECURITY__CLI_PARSER_HPP
+
+namespace eprosima {
+namespace fastdds {
+namespace examples {
+namespace security {
+
+using dds::Log;
+
+class CLIParser
+{
+public:
+
+    CLIParser() = delete;
+
+    //! Entity kind enumeration
+    enum class EntityKind : uint8_t
+    {
+        PUBLISHER,
+        SUBSCRIBER,
+        UNDEFINED
+    };
+
+    //! Configuration structure (shared for both publisher and subscriber applications)
+    struct entity_config
```

Comment:
As the subscriber does not accept a `interval` argument, better remove `uint32_t interval = 100;` from here,  create a `subscriber_config` that inherits from it and add the `uint32_t interval = 100;` there